File: src/main/java/com/connectflow/dto/UserDTO.java
package com.connectflow.dto;

import com.connectflow.model.UserRole;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserDTO {
    private UUID id;
    private String fullName;
    private String email;
    private String phone;
    private UserRole.Role role;
    private String branch;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: next/backend/src/dto/UserDTO.js
'use strict';
class UserDTO {
  constructor(user, roles = []) {
    this.id = user.id;
    this.fullName = user.fullName;
    this.email = user.email;
    this.phone = user.phone;
    this.roles = roles;
    this.createdAt = user.created_at || user.createdAt;
    this.updatedAt = user.updated_at || user.updatedAt;
  }
}
module.exports = UserDTO;
